### PR TITLE
Elasticity/Plasticity

### PR DESCRIPTION
**Pull request by boriskaus**
This adds the structs for Elasticity and Plasticity (needs more work to be actually useable)

### CODE REVIEW

File: src/Plasticity/Plasticity.jl [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    F = similar(P) 
+    @unpack_val ϕ, Ψ, C   = s
+        
+    @.   F = τII - cosd(ϕ)C - sind(ϕ)P   
+
+    return F
+end
+
+(s::DruckerPrager{_T})(P::AbstractArray{_T,N}, args...) where {_T,N} = s(P, τII; args...)
+compute_yieldfunction(s::DruckerPrager{_T}, P::AbstractArray{_T,N}, τII::AbstractArray{_T,N}, args...) where {_T,N} = s(P, τII; args...)
```
Comment (albert-de-montserrat) [inline src/Plasticity/Plasticity.jl]: No need to splat the first `args`.

File: src/Plasticity/Plasticity.jl [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+function (s::DruckerPrager{_T})(P::AbstractArray{_T,N}, τII::AbstractArray{_T,N}; kwargs...) where {_T,N}
+    F = similar(P) 
+    @unpack_val ϕ, Ψ, C   = s
+        
+    @.   F = τII - cosd(ϕ)C - sind(ϕ)P   
+
+    return F
+end
+
+(s::DruckerPrager{_T})(P::AbstractArray{_T,N}, args...) where {_T,N} = s(P, τII; args...)
```
Comment (albert-de-montserrat) [inline src/Plasticity/Plasticity.jl]: Shouldn't it be `(s::DruckerPrager{_T})(P::AbstractArray{_T,N}, τII, args) where {_T,N} = s(P, τII; args...)`

File: src/Plasticity/Plasticity.jl [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+end
+
+compute_yieldfunction(s::DruckerPrager{_T}; P::_T=zero(_T), τ_II::_T=zero(_T)) where _T = s(; P=P, τII = τII)
+
+# Calculation routine
+function (s::DruckerPrager{_T})(P::AbstractArray{_T,N}, τII::AbstractArray{_T,N}; kwargs...) where {_T,N}
+    F = similar(P) 
+    @unpack_val ϕ, Ψ, C   = s
+        
+    @.   F = τII - cosd(ϕ)C - sind(ϕ)P
```
Comment (albert-de-montserrat) [inline src/Plasticity/Plasticity.jl]: Being picky, I would write it as:
```julia
F = similar(P)
for i in eachindex(F)
     sinϕ, cosϕ =  sincosd(ϕ)
     @inbounds F[i] = τII[i] - cosϕ*C - sinϕ*P[i]   # could even use two `fma()` or `muladd()` calls here 
end
```

Comment (boriskaus) [reply src/Plasticity/Plasticity.jl]: Thanks. I think we can get rid of this function altogether (done in next PR)

**Pull request merged**